docs/source/deploy.rst

```diff
@@ -0,0 +1,179 @@
+``torch.package``. You can learn more about ``torch.package`` in the
+``torch.package`` `documentation <https://pytorch.org/docs/stable/package.html#tutorials>`_.
+
+For now, let's create a simple model that we can load and run in ``torch::deploy``.
+
+::
+
+    from torch.package import PackageExporter
+    import torchvision
+
+    # Instantiate some model
+    model = torchvision.models.resnet.resnet18()
+
+    # Package and export it.
+    with PackageExporter("my_package.pt") as e:
+        e.intern("torchvision.**")
+        e.extern("sys")
+        e.save_pickle("model", "model.pkl", model)
+
+Now, there should be a file named ``my_package.pt`` in your working directory.
+
+.. note::
+
+    Currently, ``torch::deploy`` supports only the Python standard library and
+    ``torch`` as ``extern`` modules in ``torch.package``. In the future we plan
+    to transparently support any Conda environment you point us to.
+
+
+
+Loading and running the model in C++
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's create a minimal C++ program to that loads the model.
+
+::
+
+    #include <torch/deploy.h>
+    #include <torch/script.h>
+
+    #include <iostream>
+    #include <memory>
+
+    int main(int argc, const char* argv[]) {
+        if (argc != 2) {
+            std::cerr << "usage: example-app <path-to-exported-script-module>\n";
+            return -1;
+        }
+
+        // Start an interpreter manager governing 4 embedded interpreters.
+        torch::deploy::InterpreterManager manager(4);
+
+        try {
+            // Load the model from the torch.package.
+            auto package = manager.loadPackage(argv[1]);
+            auto model = package.loadPickle("model", "model.pkl");
+        } catch (const c10::Error& e) {
+            std::cerr << "error loading the model\n";
+            return -1;
+        }
+
+        std::cout << "ok\n";
+    }
+
+This small program introduces many of the core concepts of ``torch::deploy``.
+
+An ``InterpreterManager`` abstracts over a collection of independent Python
+interpreters, allowing you to load balance across them when running your code.
+
+Using the ``InterpreterManager::loadPackage`` method, you can load a
+``torch.package`` from disk and make it available to all interpreters.
+
+Finally, ``Package::loadPickle`` allows you to retrieve specific Python objects
+from the package, like the ResNet model we saved earlier.
+
+
+Building and running the application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Assuming the above C++ program was stored in a file called, `example-app.cpp`, a
+minimal CMakeLists.txt file would look like:
+
+::
+
+    cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+    project(deploy_tutorial)
+
+    find_package(Torch REQUIRED)
+
+    add_executable(example-app example-app.cpp)
+    target_link_libraries(example-app "${TORCH_LIBRARIES}")
+    set_property(TARGET example-app PROPERTY CXX_STANDARD 14)
```

Comment:
is the CXX_STANDARD 14 related to deploy, or is that just boilerplate for any torch cmake project at this point?

Reply:
just standard, I took this from our tests